dnf/cli/commands/repoquery.py

```diff
@@ -300,7 +300,11 @@ def run(self):
         if self.opts.file:
             q = q.filter(file__glob=self.opts.file)
         if self.opts.whatprovides:
-            q = q.filter(provides__glob=[self.opts.whatprovides])
+            a = q.filter(provides__glob=[self.opts.whatprovides])
+            if len(a):
```

Comment:
If the query returns 0 packages, that it use another query. If I use statement ``if q: it answers always yes - object is there. Or am I totally wrong?